hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2013-2017 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.  See the License for the specific language
+# governing permissions and limitations under the License.
+"""A switch driver for OpenVswitch.
+"""
+import re
+import logging
+import schema
+import os
+import ast
+import subprocess
+import shlex
+
+from hil.model import db, Switch, BigIntegerType, SwitchSession
+from hil.errors import SwitchError
+
+logger = logging.getLogger(__name__)
+
+# Class layout
+# 1. Public methods:
+# 2. Private methods
+# 3. Unimplemented superclass methods.
+
+
+class Ovs(Switch, SwitchSession):
+    """ Driver for openvswitch. """
+    api_name = 'http://schema.massopencloud.org/haas/v0/switches/ovs'
+    dir_name = os.path.dirname(__file__)
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+    }
+
+    id = db.Column(
+            BigIntegerType, db.ForeignKey('switch.id'), primary_key=True
+            )
+    hostname = db.Column(db.String, nullable=False)
```

Comment:
From the description you have provided, this is actually the bridge name. If you still want to call it hostname just so that the registration call looks simpler, you might wanna put a comment explaining that. Ideally, we should call it bridgename.

Reply:
I would still like to call it hostname so as to maintain the convention followed in other switch drivers. On a local machine connecting to a switch by bridge name suffices but on a remote machine it requires IP and port info. Just fyi.

Reply:
If we do update this in the future, it will have to come with yet another migration script.